### DISCUSSION TITLE
What is the cost of building a website for selling 3D models over the internet?

### DISCUSSION THREAD

**Discussion by Icity3d** [community/community#136977] [Programming Help]
### Body

 am a 3D designer. And I create Add-ons for a 3D platform called "Blender".

I want to build a website for selling my 3D models and an Add-on with an interface for the website inside Blender (i.e., the customer can view all my 3D models from Blender but can't use them until they buy them from the website). Examples of such a website are (BlenderKit, PolyHaven, SketchFab ...)

I want these features: 1. Custom and responsive design. 2. I want to make a 3D viewer inside the webpage (i.e, the customer can interact with the 3D models inside the webpage to view it from all the angles -- something like in SketchFab 3. I want a payment portal inside the website for selling my 3D models (i.e., an e-commerce website). 4. Other designers can also upload and buy their 3D models.

So how much time and cost are estimated to build such a project concerning security, performance, and reliability measures? would it be better to ask a company to do this project for me, or to rely on freelancers? What other technical details or advice could you give me (a roadmap, a useful resource, a better approach, or anything else)?

I tried to ask chatGPT which is very frustrating (gave me a cost of up to $300k), and to search on YouTube and some other websites and I suppose according to them that a budget like $10K to $50K would do! But I am not sure, because the project is so complex and the technologies and cloud services needed to do such a project are costly.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Azymack** [comment]
The platforms you quoted are likely to use custom webgl rather than third-party plugins you can consult.
Btw, technical challenges wouldn't be big if you don't want the functionality which supports 3D model customization. You will just need to display pre-built 3D models using three.js or babylon.
Surely there are some problmes to be considered, like sharing 3D models with customers (redirect or direct download) and handling heavy data traffic (as it's gonna deal with 3D models) etc.
This kinda project which involves multi-vendor functionality, blender 3D model integration would generally take 4-6 months to launch the MVP version and you might expect beta version in 8 months.
seems like an intriguing idea.
You can connect me via email or Linkedin.

sleekj13@gmail.com
https://www.linkedin.com/in/stefan-jeremic-147bb0322
